### PR TITLE
perf(run_out): improve calculation cost of smoothPath

### PR DESCRIPTION
**Pull request by takayuki5168**
## Description

smoothPath was previously applied to all the path whose length is 200m by default which is very heavy.
This PR moved smoothPath after trimming the path.

calculation time of run_out
Before: 5 - 25ms
After: 1.5 - 3.5ms

## Tests performed

- psim
- scenario test
  - https://evaluation.tier4.jp/evaluation/reports/23a75331-6fe3-5afa-8bd3-932010c9ebf6?project_id=prd_jt

## Effects on system behavior

Nothing

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_velocity_run_out_module/src/debug.hpp [hunk 1]
```diff
@@ -35,14 +35,17 @@ class DebugValues
...
     CALCULATION_TIME = 0,
-    CALCULATION_TIME_COLLISION_CHECK = 1,
-    LATERAL_DIST = 2,
-    LONGITUDINAL_DIST_OBSTACLE = 3,
-    LONGITUDINAL_DIST_COLLISION = 4,
-    COLLISION_POS_FROM_EGO_FRONT = 5,
-    STOP_DISTANCE = 6,
-    NUM_OBSTACLES = 7,
-    LATERAL_PASS_DIST = 8,
+    CALCULATION_TIME_PATH_PROCESSING,
```
Comment (TomohitoAndo) [inline planning/behavior_velocity_run_out_module/src/debug.hpp:38]: Why did you remove the number? (`= 1`, `= 2`...)
I think it's useful when debugging with tools like plotjuggler.

Comment (takayuki5168) [reply planning/behavior_velocity_run_out_module/src/debug.hpp]: https://github.com/autowarefoundation/autoware.universe/pull/5885/commits/eb1fb4eff91bbb06a5d8ec886e8b3d9bca64f794
DONE

**Review by TomohitoAndo — APPROVED** [review]
LGTM
